File: src/com/falconraptor/utilities/sql/SQL.java
package com.falconraptor.utilities.sql;

import java.sql.*;

public class SQL {
    private static SQL instance = null;
    private Connection c = null;

    public static SQL getInstance() {
        if (instance == null) instance = new SQL();
        return instance;
    }

    public SQL connect(String sql, String u, String p, SqlType s) throws Exception {
        if (s == SqlType.MSSQL) Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
        else if (s == SqlType.MYSQL) Class.forName("com.mysql.jdbc.Driver");
        else if (s == SqlType.ACCESS) Class.forName("net.ucanaccess.jdbc.UcanaccessDriver");
        c = DriverManager.getConnection(sql, u, p);
        return this;
    }

    public Statement getStatement() throws Exception {
        return c.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_UPDATABLE);
    }

    public void close() throws SQLException {
        c.close();
    }

    public enum SqlType {MSSQL, MYSQL, ACCESS}
}

File: src/com/falconraptor/utilities/sql/SQL.scala
package com.falconraptor.utilities.sql

import java.sql.{Connection, DriverManager, ResultSet, Statement}

object SQL {
  private var c: Connection = null

  @throws(classOf[Exception]) def connect(sql: String, u: String, p: String) {
    Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver")
    c = DriverManager.getConnection(sql, u, p)
  }

  @throws(classOf[Exception]) def getStatement: Statement = c.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY)

  @throws(classOf[Exception]) def close() = c.close
}